src/components/file-indicator.js

```diff
@@ -0,0 +1,148 @@
+import clsx from 'clsx'
+import PropTypes from 'prop-types'
+import React, { forwardRef } from 'react'
+
+import { makeStyles, LinearAnimation } from '../utils/make-styles'
+
+const TimeConverter = (time) => {
+  const Minutes = Math.floor(time / 60)
+  const Seconds = time - Minutes * 60
+  if (Minutes) {
+    return `${Minutes} min${Minutes > 1 ? 's' : ''} and ${Seconds} second${Seconds > 1 || Seconds === 0 ? 's' : ''}`
+  } else {
+    return `${Seconds} second${Seconds > 1 || Seconds === 0 ? 's' : ''}`
+  }
+}
+
+const Text = (style, error, progress, time) => {
+  if (error) return (<p className={`${style.uploadedText} text-error-500`}><span className='font-bold'>Error</span>, Please Try Again Later</p>)
+  if (time) return (<p className={`${style.uploadedText} text-primary-500`}>Loaded {progress}%, {TimeConverter(time)} remaining</p>) 
+  else return (<p className={`${style.uploadedText} text-primary-500`}>Loaded {progress}%</p>)
+}
+
+const FileIndicator = forwardRef(({
+  classes, animate, direction, duration, fileName, fileSize, progress, time, pause, error = false, ...rest
+}, ref) => {
+
+  const fileIndicatorClasses = Object.freeze({ 
+    root: `h-5px w-full rounded-full bg-secondary-200 ${classes.root}`,
+    content: `h-full rounded-full bg-primary-500 ${classes.content}`,
+  })
+  const style = makeStyles({
```

Comment:
lets move makestyle class outside of the component please take a look on other component as example. We should keep a format for all the components